chain/xrp/client/client.go

```diff
@@ -0,0 +1,718 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	xc "github.com/cordialsys/crosschain"
+	xcbuilder "github.com/cordialsys/crosschain/builder"
+	"github.com/cordialsys/crosschain/chain/xrp/tx"
+	xrptxinput "github.com/cordialsys/crosschain/chain/xrp/tx_input"
+	xclient "github.com/cordialsys/crosschain/client"
+	"github.com/sirupsen/logrus"
+	"net/http"
+)
+
+// Client for XRP
+type Client struct {
+	Url        string
+	HttpClient *http.Client
+	Asset      xc.ITask
+}
+
+var _ xclient.FullClient = &Client{}
+
+// NewClient returns a new JSON-RPC Client to the XRP node
+func NewClient(cfgI xc.ITask) (*Client, error) {
+	cfg := cfgI.GetChain()
+
+	return &Client{
+		Url:        cfg.URL,
+		HttpClient: http.DefaultClient,
+		Asset:      cfgI,
+	}, nil
+}
+
+const MethodPost string = "POST"
+
+const AccountInfo string = "account_info"
+const AccountLines string = "account_lines"
+const Transaction string = "tx"
+const Ledger string = "ledger"
+const Submit string = "submit"
```

Comment:
A slightly better way to do this is:

```go
type JsonRPCMethod string
const AccountInfo JsonRPCMethod = "account_info"
const AccountLines JsonRPCMethod = "account_lines"
const Transaction JsonRPCMethod = "tx"
const Ledger JsonRPCMethod = "ledger"
const Submit JsonRPCMethod = "submit"

type LedgerIndex string

const Validated LedgerIndex = "validated"
const Current LedgerIndex = "current"

type AccountInfoRequest struct {
	Method JsonRPCMethod                  `json:"method"`
	Params []AccountInfoParamEntry `json:"params"`
}
```

Reply:
I'm also ok not typing these out, they each only tend to be used once and are pretty local.